File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:   
    def isSubtree(self, root: Optional[TreeNode], subRoot: Optional[TreeNode]) -> bool:
        if not root and not subRoot:
            return True
        elif not root or not subRoot:
            return False
        def isEqual(p, q):
            if not p and not q:
                return True
            elif not p or not q:
                return False
            return p.val == q.val and isEqual(p.left, q.left) and isEqual(p.right, q.right)
        
        return isEqual(root, subRoot) or self.isSubtree(root.left, subRoot) or self.isSubtree(root.right, subRoot)

        

File: Data Structures & Algorithms/invert-a-binary-tree/submission-4.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {TreeNode}
     */
    invertTree(root) {
        if (!root) return null;
        const stack = [root];
        while(stack.length > 0) {
            console.log(stack);
            const node = stack.pop();
            if (!node) continue;
            const temp = node.left;
            node.left = node.right;
            node.right = temp;
            stack.push(node.right);
            stack.push(node.left);
        }
        return root;
    }
}
